freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,437 @@
+'''
+This is the core logic for the FreeText XBlock
+'''
+
+import os
+import re
+
+import pkg_resources
+
+from enum import Enum
+from xblock.core import XBlock
+from xblock.fields import Scope
+from xblock.fields import String
+from xblock.fields import Integer
+from xblock.fields import Float
+from xblock.fragment import Fragment
+
+
+class Freetextresponse(XBlock):
+    #  pylint: disable=too-many-ancestors, too-many-instance-attributes
+    '''
+    Enables instructors to create questions with free-text responses.
+    '''
+    @staticmethod
+    def workbench_scenarios():
+        '''
+        Gather scenarios to be displayed in the workbench
+        '''
+        return [
+            ('FreeText XBlock',
+             '''<sequence_demo>
+                    <freetextresponse />
+                    <freetextresponse name="My First XBlock" />
+                </sequence_demo>
+             '''),
+        ]
+    display_name = String(
+        default='FreeText XBlock',
+        scope=Scope.settings,
+        help="This is the title for this question type",
+    )
+    prompt = String(
+        default='Please enter your response within this text area',
+        scope=Scope.settings,
+        help='This is the prompt students will see when '
+             'asked to enter their response',
+    )
+    weight = Integer(
+        default=1,
+        scope=Scope.settings,
+        help='This assigns an integer value representing '
+             'the weight of this problem',
+    )
+    score = Float(
+        default=0.0,
+        scope=Scope.user_state,
+        help="This is the score of the user",
+    )
+    max_attempts = Integer(
+        default=0,
+        scope=Scope.settings,
+        help='This is the maximum number of times a student '
+             'is allowed to attempt the problem',
+    )
+    display_correctness = String(
+        default='True',
+        scope=Scope.settings,
+        help='This is a flag that indicates if the indicator '
+             'icon should be displayed after a student enters '
+             'their response',
+    )
+    min_word_count = Integer(
+        default=0,
+        scope=Scope.settings,
+        help='This is the minimum number of words required for this question',
+    )
+    max_word_count = Integer(
+        default=10000,
```

Comment:
While none, would allow users to enter a response that is infinitely long, i thought putting a reasonable default value would help prevent malicious users from spamming our storage